waku/v2/protocol/waku_noise/noise_utils.nim

```diff
@@ -57,6 +57,76 @@ proc pkcs7_unpad*(payload: seq[byte], paddingSize: int): seq[byte] =
   let unpadded = payload[0..payload.high-k.int]
   return unpadded
 
+proc seqToDigest256*(sequence: seq[byte]): MDigest[256] =
+  var digest: MDigest[256]
+  for i in 0..<digest.data.len:
+    digest.data[i] = sequence[i]
+  return digest
+
+proc digestToSeq*[T](digest: MDigest[T]): seq[byte] =
+  var sequence: seq[byte]
+  for i in 0..<digest.data.len:
+    sequence.add digest.data[i]
+  return sequence
+
+# Serializes input parameters to a base64 string for exposure through QR code (used by WakuPairing)
+proc toQr*(applicationName: string, applicationVersion: string, shardId: string, ephemeralKey: EllipticCurveKey, committedStaticKey: MDigest[256]): string =
+
+  var qr: string
+  qr.add encode(applicationName) & ":"
+  qr.add encode(applicationVersion) & ":"
+  qr.add encode(shardId) & ":"
+  qr.add encode(ephemeralKey) & ":"
+  qr.add encode(committedStaticKey.data)
+
+  return qr
+
+# Deserializes input string in base64 to the corresponding (applicationName, applicationVersion, shardId, ephemeralKey, committedStaticKey)
+proc fromQr*(qr: string): (string, string, string, EllipticCurveKey, MDigest[256]) {.raises: [Defect, ValueError].} =
+
+  let values = qr.split(":")
+
+  assert(values.len == 5)
+
+  let applicationName: string = decode(values[0])
+  let applicationVersion: string = decode(values[1])
+  let shardId: string = decode(values[2])
+
+  let decodedEphemeralKey = decode(values[3]).toBytes  
+  var ephemeralKey: EllipticCurveKey 
+  for i in 0..<ephemeralKey.len:
+    ephemeralKey[i] = decodedEphemeralKey[i]
+
+  let committedStaticKey = seqToDigest256(decode(values[4]).toBytes)
+ 
+  return (applicationName, applicationVersion, shardId, ephemeralKey, committedStaticKey)
+
+# Converts a sequence or array (arbitrary size) to a MessageNametag
+proc toMessageNametag*(input: openArray[byte]): MessageNametag =
+  var byte_seq: seq[byte] = @input
+
+  # We set its length to the default message nametag length (will be truncated or 0-padded)
+  byte_seq.setLen(MessageNametagLength)
```

Comment:
Wouldn't it be simpler to loop through `input` and put `MessageNametagLength` many bytes into `messageNametag`? I mean  lines 106 and 109 can be skipped